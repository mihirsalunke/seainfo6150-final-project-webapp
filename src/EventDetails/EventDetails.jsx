import React, { useEffect, useState } from "react";
import RegisterButton from "../RegisterButton/RegisterButton.jsx";
import RegistrationForm from "../RegistrationForm/RegistrationForm.jsx";
import { isEmpty } from "lodash";
import PropTypes from "prop-types";
import styles from "./EventDetails.module.css";

const EventDetails = (props) => {

    const [registration, setRegistration] = useState(false);
    const [eventsData, setEventsData] = useState();
    const [event, setEvent] = useState();

    useEffect(() => {
        const fetchEventsData = async () => {
          const events = await fetch("http://demo0649072.mockable.io/events");
          const eventsJson = await events.json();
          setEventsData(Object.values(eventsJson));

          for(let i = 0; i < eventsJson.length; i++) {
            if(eventsJson[i].eventId === props.event) {
                setEvent(eventsJson[i]);
            }
          }
        };
    
        if (isEmpty(eventsData)) {
          fetchEventsData();
        }
    });
    
    function register() {
        setRegistration(true);
    };

    return isEmpty(event) ? <div>Fetching Event</div> : (
        <>
            {
                registration ? (
                    <RegistrationForm event={event}/>
                ) : (
                    <div className={styles.container}>
                        <div className={styles.heading}>
                            <h1>{event.title}</h1>
                        </div>
                        <div className={styles.description}>
                            <p>{event.description}</p>
                        </div>
                        <div className={styles.text}>
                            <p>On: <span>
                                <time dateTime={event.date}>{event.eventDate}</time>,   <time dateTime={event.time}>{event.eventTime}</time>
                            </span>
                            </p>
                        </div>
                        <div className={styles.text}>
                            <p>At: <span>{event.location}</span></p>
                        </div>
                        <div className={styles.organizerDetails}>
                            <div className={styles.title}><p>Organized by: <span>{event.organizerName}</span></p></div>
                            <div className={styles.text}><p>Contact: <span>{event.organizerContactNumber}</span></p></div>
                        </div>
                        <RegisterButton onClick={register}/>
                    </div>
                )
            }
        </>
    );
};

EventDetails.propTypes = {
    event: PropTypes.string.isRequired,
};

export default EventDetails;