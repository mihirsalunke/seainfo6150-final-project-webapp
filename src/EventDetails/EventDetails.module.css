@import url("https://fonts.googleapis.com/css2?family=Rowdies:wght@300&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Nerko+One&display=swap");

.container {
    background-image: linear-gradient(to right, rgb(160, 201, 226), rgb(207, 233, 248) 70%);
    max-width: 900px;
    max-height: 700px;
    border: 1px solid lightblue;
    box-shadow: 0 5px 10px 5px rgba(0, 0, 0, 0.3);
    display: flex;
    flex-flow: column;
    align-items: center;
    border-radius: 10px;
    position: fixed;
    top: 40%;
    left: 50%;
    transform: translate(-50%, -40%);
}

.heading {
    text-align: center;
    font-family: Rowdies, sans-serif;
    font-size: 1.5em;
}

.description {
    margin: 0 5%;
    padding: 0 2%;
    font-family: Rowdies, sans-serif;
}

.text {
    margin: -2% 2%;
    font-family: "Nerko One", sans-serif;
    font-size: larger;
}
.text span {
    font-family: Rowdies, sans-serif;
    font-size: 1em;
}

.title {
    margin: -2% 2%;
    font-family: "Nerko One", sans-serif;
    font-size: larger;
}
.title span {
    font-family: Rowdies, sans-serif;
}