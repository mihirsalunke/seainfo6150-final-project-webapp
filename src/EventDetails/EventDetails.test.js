import React from "react";
import EventDetails from "./EventDetails.jsx";
import { render } from "@testing-library/react";

const event = "110002";

describe("EventDetails tests", () => {
    it("renders correctly", () => {
      const { container } = render(<EventDetails event={event} />);
      expect(container).toMatchSnapshot();
    });
});