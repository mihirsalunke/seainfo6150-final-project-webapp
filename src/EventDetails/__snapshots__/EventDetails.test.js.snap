// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetails tests renders correctly 1`] = `
<div>
  <div>
    Fetching Event
  </div>
</div>
`;
